### PR TITLE
Add hexa archi to React generator

### PR DESCRIPTION
**Pull request by flcarre**
Related to #648 and #293

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/generator/client/react/core/domain/React.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class React {
+
+  private React() {}
+
+  public static List<String> dependencies() {
```
Comment (nonomoho) [inline src/main/java/tech/jhipster/lite/generator/client/react/core/domain/React.java:12]: Some of your dependencies can be move to devDependecies (all exept "react", "react-dom", "react-scripts" I think)

File: src/main/java/tech/jhipster/lite/generator/client/react/core/domain/React.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    return Map.of(
+      "start", "react-scripts start",
+      "build", "react-scripts build",
+      "test", "react-scripts test"
+    );
+    // @formatter:on
+  }
+
+  public static List<String> files() {
+    // @formatter:off
```
Comment (nonomoho) [inline src/main/java/tech/jhipster/lite/generator/client/react/core/domain/React.java:39]: Maybe useless

File: src/main/java/tech/jhipster/lite/generator/client/react/core/domain/React.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    return List.of(
+      "tsconfig.json"
+    );
+    // @formatter:on
+  }
+
+  public static Map<String, String> reactFiles() {
+    // @formatter:offProjectLocalRepository
```
Comment (nonomoho) [inline src/main/java/tech/jhipster/lite/generator/client/react/core/domain/React.java:47]: why `formatter:offProjectLocalRepository`?

File: src/main/resources/generator/dependencies/react/package.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{
+  "dependencies": {
```
Comment (nonomoho) [inline src/main/resources/generator/dependencies/react/package.json:2]: Same as above

**Comment by flcarre** [comment]
I need a maintainer to approve running workflows :)

**Comment by flcarre** [comment]
Think this one is the good one, need maintainer to run workflows again :)

**Comment by pascalgrimaud** [comment]
Closing this, following https://github.com/jhipster/jhipster-lite/issues/293#issuecomment-1048229094

**Pull request closed without merging**